c10/util/Exception.h

```diff
@@ -235,6 +235,10 @@ class C10_API LinAlgError : public Error {
   using Error::Error;
 };
 
+class C10_API OutOfMemoryError : public Error {
```

Comment:
If we want to keep the cuda there, we could put this in a `torch::cuda::` namespace?
But on the other hand, maybe we also want our CPU allocator to raise this?

Reply:
I think it is ok to generalize this. Previous to now it really was only used in some C++ tests to check that OOMs were being raised in certain situations. I can see a future where we extend the exception with a device attribute to describe which thing ran out of memory if we really needed the distinction.

Reply:
Sounds good!